pkg/domainservice/adapterInterfaces.go

```diff
@@ -98,6 +109,12 @@ func (e *NotFoundError) Unwrap() error {
 	return e.WrappedError
 }
 
+// NewInternalError creates an InternalError with a given message. The wrapped error may be nil. The error message must
+// omit the fmt.Errorf verb %w because this is done by InternalError.Error().
+func NewInternalError(wrappedError error, message string, msgArgs ...interface{}) *InternalError {
```

Comment:
Nicely done adding a constructor for this!

Reply:
Thanks